addons/analytic/models/analytic_account.py

```diff
@@ -1,15 +1,56 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import api, fields, models
+from odoo import api, fields, models, exceptions, _
 
 
+class AccountAnalyticDistribution(models.Model):
+    _name = 'account.analytic.distribution'
+    _rec_name = 'account_id'
+
+    account_id = fields.Many2one('account.analytic.account', string='Analytic Account', required=True)
+    percentage = fields.Float(string='Percentage', required=True, default=100.0)
+    name = fields.Char(string='Name', related='account_id.name')
+    tag_id = fields.Many2one('account.analytic.tag', string="Parent tag")
+
+    @api.onchange('percentage')
+    def _onchange_percentage(self):
+        if self.percentage > 100 or self.percentage <= 0:
+            return {'warning': {
+                    'title' : _("Invalid percentage"),
+                    'message': _("The percentage must be between 0% and 100%")
+                }
+            }
+
+    @api.multi
+    def name_get(self):
+        res = []
+        for distribution in self:
+            res.append((distribution.id, distribution.account_id.name))
+        return res
+
+    @api.model
+    def name_search(self, name='', args=None, operator='ilike', limit=100):
+        if args is None:
+            args = []
+        domain = args + ['|', ('account_id', operator, name)]
```

Comment:
Could you enlight me please, would something like this would be better.
```python
    @api.model
    def name_search(self, name, args=None, operator='ilike', limit=100):
        if name:
            recs = self.search([('account_id', operator, name)] + (args or []), limit=limit)
            return recs.name_get()
        return super(AccountAnalyticDistribution, self).search(domain, limit=limit).name_get()
```